src/variables.jl

```diff
@@ -99,6 +99,10 @@ Base.isequal(a::Constant, b::Constant) = a.value == b.value
 
 Base.convert(::Type{Expr}, c::Constant) = c.value
 
+# For dictionary lookup
+Base.hash(v::Variable,h) = hash(Symbol(v),h)
+
```

Comment:
```suggestion
Base.hash(v::Variable, u::UInt64) = hash(Symbol(v), u)
```
seems to avoid the method ambiguity, though there really should be a better solution.

Reply:
yeah, that's why I stopped this WIP. There's a real design decision hiding in here for handling some of the cases.